core_chat_data/src/commonMain/kotlin/net/thechance/mena/core_chat/data/di/networkModule.kt

```diff
@@ -0,0 +1,16 @@
+package net.thechance.mena.core_chat.data.di
+
+import io.ktor.client.engine.HttpClientEngineConfig
+import io.ktor.client.engine.HttpClientEngineFactory
+import net.thechance.mena.core_chat.data.network.createHttpClient
+import org.koin.core.qualifier.named
+import org.koin.core.scope.Scope
+import org.koin.dsl.module
+
+internal val networkModule = module {
+    single(named("baseUrl")) { "http://10.0.2.2:8080" } // TODO change to real url in production
+    single { createHttpClientEngine() }
+    single { createHttpClient(get(named("baseUrl")), get()) }
+}
+
+expect fun Scope.createHttpClientEngine(): HttpClientEngineFactory<HttpClientEngineConfig>
```

Comment:
```suggestion
expect fun createHttpClientEngine(): HttpClientEngineFactory<HttpClientEngineConfig>
```
We don't need the scope extension here it's just returning the platform specific Engine Object